flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -303,7 +302,7 @@ ANY, and(logical(LogicalTypeRoot.BOOLEAN), LITERAL)
                     .name("ARRAY_MAX")
                     .kind(SCALAR)
                     .inputTypeStrategy(arrayFullyComparableElementType())
-                    .outputTypeStrategy(new ArrayElementOutputTypeStrategy())
+                    .outputTypeStrategy(forceNullable(SpecificTypeStrategies.ARRAY_ELEMENT))
```

Comment:
Hi, @liuyongvs I also have question about this, you use forceNullable to force INT NOT NULL -> INT.
But the SpecificTypeStrategies.ARRAY_ELEMENT return type is INT NOT NULL, does it?

Reply:
@hanyuzheng7 it can be reuse for others.